base/tps/src/org/dogtagpki/server/tps/TPSSubsystem.java

```diff
@@ -131,8 +131,8 @@ public void init(ISubsystem owner, IConfigStore config) throws EBaseException {
 
         FileConfigStore defaultConfig = new FileConfigStore("/usr/share/pki/tps/conf/CS.cfg");
 
-        uiTransitions = loadAndValidateTokenStateTransitions(
-                defaultConfig, cs, TPSEngine.CFG_TOKENDB_ALLOWED_TRANSITIONS);
+        uiTransitions = loadTokenStateTransitions(
```

Comment:
For my understanding, what's the difference between UI Transitions and Operation Transitions, and why is one validated on loading (Operation) but the other isn't (UI)?

Reply:
I think UI transitions are transitions that can be done in TPS database without involving the token itself (e.g. suspending the token). On the other hand transitions due to token operations requires the actual token (e.g. installing applet, certs, keys).

Generally UI transitions are customizable according to the company's policy, but operation transitions are not since usually it has to be supported by the code. The TPS UI should automatically reflect the actual transitions configured in the CS.cfg.

Reply:
Fair enough, thanks for the response. :)